CMakeLists.txt

```
@@ -44,13 +45,13 @@ if(BUILD_TESTING)
   # ament_lint_auto_find_test_dependencies()
   
   find_package(ament_cmake_gtest REQUIRED)
-  ament_add_gtest(${PROJECT_NAME}_test test/heartbeat_test.cpp)
+  ament_add_gtest(${PROJECT_NAME}_test test/common_node_test.cpp)
```

Ich hätte die test nach geprüfter Oberfunktionalität getrennt, also eine zweite Sourcefile erstellt.
Die beiden könnte man dann in test/common_node/... packen.
```suggestion
  ament_add_gtest(${PROJECT_NAME}_test
    test/common_node/finish_function_test.cpp
    test/common_node/heartbeat_test.cpp
  )
```

---

Ich habs jetzt aufgeteilt in mehrere Dateien. Ich hatte davor das Problem, dass er mir dann natürlich wegen mehreren Mains gemeckert hat und ich wusste nicht, dass einmal main reicht. Jetzt läufts aber.